src/shared.ts

```diff
@@ -96,26 +113,30 @@ export const loadScript = (
         console.warn(EXISTING_SCRIPT_MESSAGE);
       } else if (!script) {
         script = injectScript(params);
+      } else if (script) {
+        // remove event listeners
+        script.removeEventListener('load', onLoad(resolve, reject));
+        script.removeEventListener('error', onError(reject));
+
+        // if script exists, but we are reloading due to an error,
+        // reload script to trigger 'load' event
+        script.parentNode?.removeChild(script);
+        script = injectScript(params);
       }
 
-      script.addEventListener('load', () => {
-        if (window.Stripe) {
-          resolve(window.Stripe);
-        } else {
-          reject(new Error('Stripe.js not available'));
-        }
-      });
-
-      script.addEventListener('error', () => {
-        reject(new Error('Failed to load Stripe.js'));
-      });
+      script.addEventListener('load', onLoad(resolve, reject));
+
+      script.addEventListener('error', onError(reject));
```

Comment:
Hmmmm, I forgot that `resolve` and `reject` need to be passed as arguments. You'll need to persist the listeners at the module level alongside the `let stripePromise` declaration. E.g.

```typescript
let stripePromise = /* initializer */;
let onLoadListener = null;
let onErrorListener = null;

// loadScript
onLoadListener = onLoad(resolve, reject);
onErrorListener = onError(reject);
script.addEventListener('load', onLoadListener);
script.addEventListener('error', onErrorListener;
```

When calling `removeEventListener()`, you would then reference the `onLoadListener` and `onErrorListener` variables already defined. Does that make sense?

Reply:
Okay and just to clarify `onLoadListener = onLoad(resolve, reject);` the instances of `onLoadListener` would stay equivalent even when it's assigned again on retry?